### PR TITLE
Fix summons & familiars

### LINKED ISSUE #1250: Summons with targetdistance running away from target
Summons that are supposed to attack from distance (targetdistance=">0") are running away "infinitely" (until target is lost for their master), if attacked monster target them.
[Vid showing that problem.](https://gfycat.com/importantfastdesertpupfish)

To reproduce:
1. Summon creature that attack from distance (emberwing, minotaur archer etc)
2. Attack hostile monster
3. Let that monster target your summon instead of you
4. Your summon is running away like crazy

Expected behavior:
Your summon is walking one tile every 2 sec. [Video showing that](https://youtu.be/68AVCNMZ8Bs?t=490) - skullfrost is slowly walking away instead of running.

I tried to fix that myself for a while, but failed as my attempt didnt work.

Code to look at:
https://github.com/opentibiabr/otservbr-global/blob/9965cd24f4ce5b9851e6e95dfc2d7e1cbde3b574/src/creature.cpp#L895
https://github.com/opentibiabr/otservbr-global/blob/9965cd24f4ce5b9851e6e95dfc2d7e1cbde3b574/src/monster.cpp#L1361
https://github.com/opentibiabr/otservbr-global/blob/ea1d34d4a6b8a88a4eec6712c082a430eb4a4cde/src/monster.cpp#L805

### LINKED ISSUE #2058: BUG: Familiars wont walk over fields
#### Describe the bug
After being summoned the familiar wont walk over fire/poison/energy field

#### How to Reproduce?
Summon a familiar and shot fire/poison/energy wall/bomb

#### Logs or errors
no errors shown

#### Expected behavior
Familiar should walk over it w/o problem.

#### Screenshots
ezgif-2-68403e63eb05_1
Sorry for the low quality but wasnt able to make it lesser than 10mb :s , anyway, you're still able to get the thing

#### Environment
- OS (e.g. Windows): Debian 10
- OTServ Version (e.g. commit): Develop
- Tibia Version (e.g. 12.52): 12.51

### LINKED ISSUE #2143: BUG: Summon disapears
#### Describe the bug
If players gets surrounded by monsters summon disappears 

#### How to Reproduce?
Get surrounded by monsters while having summon ~grovebeast ~

#### Expected behavior
Summon should not disappear

### LINKED ISSUE #2156: BUG: familiars
I found an error in the update of the familiars, when a player makes his familiar all others that enter the server later will have the same familiar with them regardless of the level or vocation

### LINKED ISSUE #2179: FEATURE: Familiars fly in front of me
#### Describe your feature request
Familiars fly in front of me

#### Describe the solution you'd like
familiar should wait and follow the master, not in front of him (i checked at global)

#### Screenshots
image

### LINKED ISSUE #2257: Server Crash After changing outfit in dawnport
Server just closes when i try to change my outfit on dawnport after i pass a vocation change thing.

Get no error's.

### PR DESCRIPTION
**Pull request by dex-89**
- [X] Changed storage name
- [X] Fixed event 'onLook'
- [X] Added missing line (#2158) : by @hyresu 
- [X] Fixed familiars disappearing
- [X] Player can attack his familiar / summon
- [X] Fixed event 'onDeath'
- [X] Disappearance communication
- [X] Familiars fly in front of player: by @hyresu 
- [X] Summons with targetdistance running away from target (https://github.com/opentibiabr/otservbr-global/issues/1250): by @hyresu
- [X] Fixed Dawnport crash @hyresu 
- [X] Fixed Avoid unnecessary heap allocations in function "ProtocolGame::sendOutfitWindow()" @SaiyansKing 
- [X] Fixed: `he gets trapped by monsters the summon disappear`

### CODE REVIEW

**Comment by helltheus** [comment]
https://user-images.githubusercontent.com/68662258/103376659-131e6c00-4abc-11eb-8c1b-3103e28d41cb.mp4

File: data/spells/scripts/summon/familiars.lua [hunk 1]
```diff
@@ -34,31 +45,34 @@ function onCastSpell(player, variant)
...
-    mySummon:reload()
-    mySummon:registerEvent("SummonDeath")
+    player:addSummon(myFamiliar)
+    myFamiliar:setOutfit({lookType = player:getFamiliarLooktype()})
+    myFamiliar:reload()
+    myFamiliar:registerEvent("FamiliarDeath")
 
-    local deltaSpeed = math.max(player:getBaseSpeed() - mySummon:getBaseSpeed(), 0)
-    mySummon:changeSpeed(deltaSpeed)
+    local deltaSpeed = math.max(player:getBaseSpeed() - myFamiliar:getBaseSpeed(), 0)
```
Comment (hyresu) [inline data/spells/scripts/summon/familiars.lua]: TibiaWiki states "Their speed is the same as their master's.", wouldn't this formula don't take in count the speed buffs?

Comment (dex-89) [reply data/spells/scripts/summon/familiars.lua]: get**Base**Speed = player base speed

**Review by Risk-CPU — APPROVED** [review]

**Review by paulorob-dev — APPROVED** [review]

**Review by majestyotbr — APPROVED** [review]

**Comment by yatozin** [comment]
The message to say summon only has 1 minute left still appears if summon has died

And summons also still disappear

**Comment by yatozin** [comment]
The disappearing summon appears to happen when the player gets fully boxed by monsters
edit happens even when not :s

**Comment by hyresu** [comment]
image

Not happening for me, could you test it on a fresh develop branch?

**Comment by gynho** [comment]
> imagem
> 
>Not happening for me, could you test it on a fresh develop branch?

Hi! I tested with rotworms and nothing happened but testing with dragons and asuras the summon disappeared when it was trapped. Maybe it's gone for pushing monsters? I can't say ... but it's gone!

**Comment by helltheus** [comment]
I have test too, when he gets trapped by monsters the summon disappear, it doesn't always happen.

**Comment by yatozin** [comment]
> image
> 
> Not happening for me, could you test it on a fresh develop branch?

tested again on base test were at asuras and summon needs to be on the outside of the box, but it also happens when not boxed so not sure why

File: data/spells/scripts/summon/familiars.lua [hunk 2]
```diff
@@ -34,31 +45,33 @@ function onCastSpell(player, variant)
...
-    player:addSummon(mySummon)
-    mySummon:setOutfit({lookType = player:getFamiliarLooktype()})
-    mySummon:reload()
-    mySummon:registerEvent("SummonDeath")
-
-    local deltaSpeed = math.max(player:getBaseSpeed() - mySummon:getBaseSpeed(), 0)
-    mySummon:changeSpeed(deltaSpeed)
+    player:addSummon(myFamiliar)
+    myFamiliar:setOutfit({lookType = player:getFamiliarLooktype()})
+    myFamiliar:reload()
```
Comment (gpedro) [inline data/spells/scripts/summon/familiars.lua]: what is that for?

Comment (dex-89) [reply data/spells/scripts/summon/familiars.lua]: I dont know. That was before I started reworking

**Comment by helltheus** [comment]
I have this error
image

**Comment by hyresu** [comment]
> I have this error
> image

Replace `return combat:execute(player, variant)` with `return true`

Or simply pull the last change I made (:

**Comment by yatozin** [comment]
Seems to be working now

**Review by majestyotbr — APPROVED** [review]

**Review by hyresu — APPROVED** [review]

**Comment by Plankada** [comment]
look it was 4 hours putting the family and the same error did not stop, can you help?
WhatsApp Image 2021-01-16 at 18 30 19

WhatsApp Image 2021-01-16 at 18 30 49

**Comment by dex-89** [comment]
> look it was 4 hours putting the family and the same error did not stop, can you help?
> WhatsApp Image 2021-01-16 at 18 30 19
> 
> WhatsApp Image 2021-01-16 at 18 30 49

are you on discord?

https://discordapp.com/invite/3NxYnyV

Find me there and send me the packed sources. I will check what is happening and where this error comes from. I have the same nickname as here.

**Pull request merged**